src/components/TagPicker/index.js

```diff
@@ -57,6 +46,16 @@ function TagPicker({selectedTag, tag, policyTags, policyRecentlyUsedTags, onSubm
         return tags;
     }, [selectedOptions, policyTagList, shouldShowDisabledAndSelectedOption]);
 
+    const initialFocusedIndex = useMemo(() => {
+        if (isTagsCountBelowThreshold && selectedOptions.length > 0) {
+            return _.chain(enabledTags)
```

Comment:
I see the fix is replacing `policyTagList` with `enabledTags` but `shouldShowDisabledAndSelectedOption` = false on new request so not affected as early returning [here](https://github.com/Expensify/App/pull/31652/files#diff-7d3f6fc1fc99f80e69795d6996b7b67a94b4feb95af2049126abe03b057741faR42). This change comes into effect on edit request case.
Am I missing anything?

Reply:
Correct, the issue only affects the edit case

Reply:
But the repro step in https://github.com/Expensify/App/issues/31578 is new request case. Also video states that.

Reply:
The step says `Open the tags label again`

Reply:
> The step says `6. Open the tags label again`, which is after we already selected one in `5. Select any option`

That doesn't matter. Both cases are from `MoneyRequestTagPage`.
`shouldShowDisabledAndSelectedOption` is enabled only on `EditRequestTagPage`.

Reply:
Right, but as you noticed the `enabledTags` order is different when we selected something, which throws off the indexes and it's what I described in the OP

Reply:
The bug is still reproducible for me on this PR when `shouldShowDisabledAndSelectedOption` = true.

https://github.com/Expensify/App/blob/da549c0a45edb23fbe9616a9ce21915f291eabe4/src/components/TagPicker/index.js#L56
Changing this line to
`const tags = _.filter(policyTagList, (policyTag) => policyTag.enabled || selectedNames.includes(policyTag.name));`

fixes the issue

Reply:
Ah you're right. Pushed your suggestion

Reply:
hmm I'm still seeing the issue on the latest update

Reply:
Don't revert your original change 🙂 
We need both fixes

Reply:
Seems to be fixed now